agent/metrics_test.go

```diff
@@ -432,3 +432,193 @@ func TestHTTPHandlers_AgentMetrics_CACertExpiry_Prometheus(t *testing.T) {
 	})
 
 }
+
+func TestHTTPHandlers_AgentMetrics_WAL_Prometheus(t *testing.T) {
+	skipIfShortTesting(t)
+	// This test cannot use t.Parallel() since we modify global state, ie the global metrics instance
```

Comment:
_Leaving this comment with some gaps in context, so apologies in advance._

I understand why we're doing this, and it seems to be an existing pattern in this file, but modifying global state is usually bad. Not necessarily one for this PR, but could we just clone/create new metrics instances and operate on these instead?

Reply:
Great callout. This is a shortcoming of go-metrics in general. I'd _love_ to move away from go-metrics to something more modern but that comes with a lot of compatibility issues since all our metrics config and output depends on quirks of how go-metrics has always done things. All HashiCorp products are also built on it so it is a big change to go changing how go-metrics works.

If we wanted to fix just the global state part it would at least mean updating every instrumentation point in our code that calls `metrics.*()` functions to instead call a method on a specific instance which is part of the agent dependencies and plumbed though all the other modules and layers. I'd _love_ to get there one day but it's not been something we've chosen to bite off just yet!